manticore/platforms/evm.py

```diff
@@ -261,101 +265,286 @@ def write(self, address, value):
+            self._pops = pops
+            self._pushes = pushes
+            self._fee = fee
+            self._description = description
+            self._operand = operand           #Immediate operand if any
+            if operand_size != 0 and operand is not None:
+                    mask = (1<<operand_size*8)-1
+                    if ~mask & operand:
+                        raise ValueError("operand should be %d bits long"%(operand_size*8))
+            self._offset=offset
+
+        def __eq__(self, other):
+            ''' Instructions are equal if all features match '''
+            return self._opcode == other._opcode and\
+            self._name == other._name and\
+            self._operand == other._operand and\
+            self._operand_size == other._operand_size and\
+            self._pops == other._pops and\
+            self._pushes == other._pushes and\
+            self._fee == other._fee and\
+            self._offset == other._offset and\
+            self._description == other._description 
+
+        def __repr__(self):
+            output = 'Instruction(0x%x, %r, %d, %d, %d, %d, %r, %r, %r)'%(self._opcode, self._name, self._operand_size, self._pops, self._pushes, self._fee, self._description, self._operand, self._offset)
+            return output
+
+
+        def __str__(self):
+            output = self.name + (' 0x%x'%self.operand if self.has_operand else '')
+            return output
+
+        @property
+        def opcode(self):
+            ''' The opcode as an integer ''' 
+            return self._opcode
+
+        @property
+        def name(self):
+            ''' The instruction name/mnemonic ''' 
+            if self._name == 'PUSH':
+                return 'PUSH%d'%self.operand_size
+            elif self._name == 'DUP':
+                return 'DUP%d'%self.pops
+            elif self._name == 'SWAP':
+                return 'SWAP%d'%(self.pops-1)
+            elif self._name == 'LOG':
+                return 'LOG%d'%(self.pops-2)
+            return self._name
+            
+        def parse_operand(self, buf):
+            ''' Parses an operand from buf 
+                :param buf: a buffer
+                :type buf: iterator/generator/string
+            '''
+            buf = iter(buf)
+            try:
+                operand = 0
+                for _ in range(self.operand_size):
+                    operand <<= 8
+                    operand |= ord(next(buf))
+                self._operand = operand
+            except StopIteration:
+                raise Exception("Not enough data for decoding")
+
+        @property
+        def operand_size(self):
+            ''' The immediate operand size '''
+            return self._operand_size
+
+        @property
+        def has_operand(self):
+            ''' Flag that indicates if the instruction uses an immediate operand'''
+            return self.operand_size > 0
+
+        @property
+        def operand(self):
```

Comment:
we have a lot of properties here that simple return an underscore attribute. is this necessary? can we simply use public class attributes? we can document them with Sphinx `:ivar:`

Reply:
The idea of a property returning _some_attribute is to make the some_attribute kind of read_only.
Can we dosument @properties with :ivar: ?